app/uk/gov/hmrc/cipphonenumberverification/services/PasscodeService.scala

```diff
@@ -19,25 +19,27 @@ package uk.gov.hmrc.cipphonenumberverification.services
 import play.api.Logging
 import uk.gov.hmrc.cipphonenumberverification.models.{Passcode, PhoneNumber}
 import uk.gov.hmrc.cipphonenumberverification.repositories.PasscodeCacheRepository
-import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.mongo.cache.DataKey
 
 import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
 
-class PasscodeService @Inject()(passcodeCacheRepository: PasscodeCacheRepository) extends Logging {
-  def persistPasscode(phoneNumber: PhoneNumber)(implicit hc: HeaderCarrier) = {
+class PasscodeService @Inject()(passcodeCacheRepository: PasscodeCacheRepository)
+                               (implicit ec: ExecutionContext) extends Logging {
+  def persistPasscode(phoneNumber: PhoneNumber): Future[Passcode] = {
     logger.debug(s"Storing passcode in database for ${phoneNumber.phoneNumber}")
     val otp = OtpService.otpGenerator
     val passcode = Passcode(phoneNumber.phoneNumber, otp)
-    passcodeCacheRepository.persistPasscode(phoneNumber, passcode)
+    passcodeCacheRepository.put(phoneNumber.phoneNumber)(DataKey("cip-phone-number-verification"), passcode)
+      .map(_ => passcode)
   }
 
-  def retrievePasscode(passcode: Passcode) = {
+  def retrievePasscode(passcode: Passcode): Future[Option[Passcode]] = {
```

Comment:
Most times not necessary to explicitly write down the return type, i suppose its down to choice

Reply:
intellij suggests adding types for all public members

Reply:
i agree with IntelliJ, we should have it for the public ones